### ISSUE TITLE
VOS Evaluation Results Mismatch with Reported Numbers

### ISSUE THREAD

**Issue by Mark20010319** [yangchris11/samurai#119]
Hello, thank you for your great work and for open-sourcing the SAMURAI codebase!
While reproducing the evaluation results for video object segmentation (VOS) tasks, I found a discrepancy between my reproduced results and those reported in [Issue #104](https://github.com/yangchris11/samurai/issues/104), as well as those in the recent preprint:
https://arxiv.org/pdf/2505.00739v1
I cloned the repo and ran the evaluation directly without modification, yet my results are closer to those in the paper above rather than Issue #104. I'm wondering if there might be any missing configuration or settings that I overlooked?
Below are my reproduced results for SA-V validation and test sets:
| Model        | SA-V val J&F | J    | F    | SA-V test J&F | J    | F    |
|--------------|--------------|------|------|----------------|------|------|
| SAMURAI-T    | 76.2         | 72.3 | 80.1 | 77.4           | 73.4 | 81.5 |
| SAMURAI-S    | 77.4         | 73.6 | 81.3 | 77.4           | 73.6 | 81.3 |
| SAMURAI-B+   | 76.9         | 73.1 | 80.7 | 78.5           | 74.6 | 82.5 |
| SAMURAI-L    | 79.8         | 75.9 | 83.6 | 80.0           | 76.2 | 83.9 |

Would you mind clarifying whether any specific settings or environment details are necessary to match the results from Issue #104?
Thanks again for your contributions!

**Comment by yangchris11** [comment]
Hi @Mark20010319 

Thank you for your interest in our work. The results in #104 are from a distinct set of parameters specifically select for VOS task as we observe that VOT and VOS tasks are quite different in terms of characteristic.

I am not able to find the specific set of numbers right away (I do found the [results.csv files](https://drive.google.com/drive/folders/1CI9RsDnr2RUZ5Y1bpNfZgJvJc42qJixA?usp=sharing) along with the mask results we reported upon). But the best shot will be changing the parameters here in the config file:
```
  # SAMURAI
  samurai_mode: true
  stable_frames_threshold: 15
  stable_ious_threshold: 0.3
  min_obj_score_logits: -1
  kf_score_weight: 0.15
  memory_bank_iou_threshold: 0.5
  memory_bank_obj_score_threshold: 0.0
  memory_bank_kf_score_threshold: 0.0
```

From what I recall, I experimented with different `stable_frames_threshold` and `min_obj_score_logits` values. That said, these were quick experiments done under time constraints, and more careful tuning could likely yield even better results.

**Comment by yangchris11** [comment]
Also, please expect some delay in this as I’m currently caught up with other commitments. 

I’m open to discussing any specific questions or ideas you have regarding the parameter tuning, evaluation setup, or general VOS/VOT differences—feel free to reach out anytime at [cycyang@uw.edu](mailto:cycyang@uw.edu). :)
